covertreemap/percent_coverage.go

```diff
@@ -0,0 +1,37 @@
+package covertreemap
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/nikolaydubina/treemap"
+)
+
+func AddCoveragePercentageToName(tree *treemap.Tree) {
+	for path, node := range tree.Nodes {
+		parts := strings.Split(node.Path, "/")
+		if len(parts) == 0 {
+			continue
+		}
+
+		var builder strings.Builder
+
+		if path == tree.Root {
+			builder.WriteString(node.Name)
+		} else {
+			builder.WriteString(parts[len(parts)-1])
+		}
+
+		builder.WriteString(" ")
+		builder.WriteString(strconv.FormatFloat(node.Heat*100, 'f', 1, 64))
+		builder.WriteString("%")
+
+		tree.Nodes[path] = treemap.Node{
+			Path:    node.Path,
+			Name:    builder.String(),
+			Size:    node.Size,
+			Heat:    node.Heat,
+			HasHeat: node.HasHeat,
+		}
```

Comment:
will this work?
```suggestion
		tree.Nodes[path] = treemap.Node{
			Path:    node.Path,
			Name:    node.Name + " " + strconv.FormatFloat(node.Heat*100, 'f', 1, 64),
			Size:    node.Size,
			Heat:    node.Heat,
			HasHeat: node.HasHeat,
		}
```

Reply:
both options are working, but with different concatenation performance

Reply:
yeah, thats true. strings builder avoids memory copy for strings.

probably, you would want to grow it to avoid memory allocation. but in this application this is okay.

<img width="912" alt="image" src="https://github.com/user-attachments/assets/bb2d0291-4422-4583-a5b0-b54c9666255a">

ideally, we would use Writer/Marshaller pattern, for whole project. maybe another time.